plugins/mac-virtualcam/src/camera-extension/OBSCameraDeviceSource.swift

```diff
@@ -0,0 +1,291 @@
+//
+//  OBSCameraDeviceSource.swift
+//  mac-camera-extension
+//
+//  Created by Sebastian Beckmann on 30.09.22.
+//  Changed by Patrick Heyer on 16.10.22.
+//
+
+import AppKit
+import CoreMediaIO
+import Foundation
+import IOKit.audio
+import os.log
+
+let OBSCameraFrameRate: Int = 60
+
+class OBSCameraDeviceSource: NSObject, CMIOExtensionDeviceSource {
+    private(set) var device: CMIOExtensionDevice!
+
+    private var _streamSource: OBSCameraStreamSource!
+    private var _streamSink: OBSCameraStreamSink!
+
+    private var _streamingCounter: UInt32 = 0
+    private var _streamingSinkCounter: UInt32 = 0
+
+    private var _timer: DispatchSourceTimer?
+
+    private let _timerQueue = DispatchQueue(
+        label: "timerQueue",
+        qos: .userInteractive,
+        attributes: [],
+        autoreleaseFrequency: .workItem,
+        target: .global(qos: .userInteractive)
+    )
+
+    private var _videoDescription: CMFormatDescription!
+    private var _bufferPool: CVPixelBufferPool!
+    private var _bufferAuxAttributes: NSDictionary!
+
+    private var _whiteStripeStartRow: UInt32 = 0
+    private var _whiteStripeIsAscending: Bool = false
```

Comment:
These look like a leftover of the code template for the camera extension that this uses which I didn't notice but aren't needed anymore.